dropshot/src/handler.rs

```diff
@@ -129,32 +128,11 @@ impl<Context: ServerContext> RequestContext<Context> {
 
         Ok(pag_params
             .limit
-            /*
-             * Convert the client-provided limit from a NonZeroU64 to a
-             * usize.  That's because internally, we want the limit to be a
-             * "usize" so we can use functions like `iter.take()` with it (as an
-             * example).  We could put "usize" in the public interface, but that
-             * would cause the server's exported interface to change when it was
-             * built differently, although that's arguably correct.  Instead, we
-             * essentially validate here that the client gave us a value that we
-             * can support.
-             */
-            .map(|limit_nzu64| usize::try_from(limit_nzu64.get()))
-            .transpose()
-            .map_err(|_| {
-                HttpError::for_bad_request(
-                    None,
-                    String::from("unsupported pagination limit: too large"),
-                )
-            })?
             /*
              * Compare the client-provided limit to the configured max for the
              * server and take the smaller one.
              */
-            .map(|limit_usize| {
-                let limit_nzusize = NonZeroUsize::new(limit_usize).unwrap();
-                min(limit_nzusize, server_config.page_max_nitems)
-            })
+            .map(|limit_nzu32| min(limit_nzu32, server_config.page_max_nitems))
```

Comment:
Seems like we could remove the type suffix here, since there's no conversion?

Reply:
Good point.  Fixed in ed77606ea778827f4fd456cf622f4be5f4ea170d.